lib/kb_mash/mash_utils/MashUtils.py

```diff
@@ -93,9 +110,42 @@ def sketch_service_query(self, input_upas, n_max_results, search_db):
                 }
             }
             print("Processing input: %s" % input_name)
-            resp = requests.post(url=sketch_url, data=json.dumps(payload),headers={
-                'content-type':"application/json-rpc",'Authorization':self.auth_token},
-                timeout=600)
+            attempts = 1
+            max_attempts = 5
+            while True:
+                # there's really no good way to test retries w/o a ton of work
+                start = time.time()
+                try:
+                    resp = requests.post(url=sketch_url, data=json.dumps(payload),headers={
+                        'content-type':"application/json-rpc",'Authorization':self.auth_token},
+                        timeout=600)
+                except RequestException as e:
+                    if attempts > max_attempts:
+                        # dunno if e.args[0] is right, no docs  
+                        # https://requests.readthedocs.io/en/master/api/#exceptions
+                        raise SketchException('Sketch service communications failed: ' +
+                                              str(e.args[0])) from e
+                    log('Sketch service comms failed after ' + str(time.time() - start) + 's')
+                    logerr(e)
+                else:
+                    if not resp.ok:
+                        # could try deserializing from json, YAGNI for now
+                        if attempts > max_attempts:
+                            raise SketchException(
+                                'Recieved bad response from the sketch service: {}\n{}'.format(
+                                    resp.status_code, resp.text))
+                        log('Bad response from Sketch service after ' +
+                            str(time.time() - start) + 's')
+                        log('Error from sketch service: {}\n{}'.format(
+                            resp.status_code, resp.text))
+                    else:
+                        break
```

Comment:
Does this `break` effect only the try/except construct and not the while loop? I'm not sure, but it might be the case that in python `break` will exit the try/except: https://docs.python.org/3/tutorial/errors.html#defining-clean-up-actions

Reply:
The tests that exist pass and produce results, so it must break out of the while loop, otherwise the tests would hang forever